src/dimred/lsi/script.py

```diff
@@ -0,0 +1,105 @@
+import scanpy as sc
+import muon as mu
+import mudata as md
+import logging
+from anndata import AnnData
+from sys import stdout
+import numpy as np
+import sys
+
+## VIASH START
+par = {
+    "num_components": 50, # number of components to calculate with SVD
+    "scale_embeddings": True, # scale embeddings to zero mean and unit variance
+    "modality": "atac", # on which modality the LSI should be run 
+    "layer": None, # on which layer to run the LSI, if None, will run it on anndata.X 
+    "var_input": "highly_variable", # column in anndata.var of the highly variable features 
+    #parameters for the output: 
+    "overwrite": True, 
+    "obsm_output": "X_lsi", # name of the slot in .obsm
+    "varm_output": "LSI", #name of the slot in .varm
+    "uns_output": "lsi", # name of the slot in .uns
+    "output": "output.h5mu", #path to the output file 
+    "output_compression": "gzip"
+}
+## VIASH END
+
+
+sys.path.append(meta["resources_dir"])
+# START TEMPORARY WORKAROUND setup_logger
+# reason: resources aren't available when using Nextflow fusion
+# from setup_logger import setup_logger
+def setup_logger():
+    import logging
+    from sys import stdout
+
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    console_handler = logging.StreamHandler(stdout)
+    logFormatter = logging.Formatter("%(asctime)s %(levelname)-8s %(message)s")
+    console_handler.setFormatter(logFormatter)
+    logger.addHandler(console_handler)
+
+    return logger
+# END TEMPORARY WORKAROUND setup_logger
+logger = setup_logger()
```

Comment:
Usually, code is wrapped in the `main()` function. See an example here: https://github.com/openpipelines-bio/openpipeline/blob/main/src/integrate/scvi/script.py

Reply:
The code of the other dimreds, i.e. UMAP and PCA is also not wrapped into a main: [UMAP](https://github.com/openpipelines-bio/openpipeline/blob/main/src/dimred/umap/script.py), [PCA](https://github.com/openpipelines-bio/openpipeline/blob/main/src/dimred/pca/script.py)

Reply:
I think we use both styles in openpipelines. Functionally, using `if __name__ == "__main__"` matters when the code is being imported as a python package (which is never the case for components). So in OpenPipelines I think the usage of `main()` depends on whether or not the script has become large enough for the code to be split up in several functions. When this point is reached is of course completely subjective 🤷 😄